### PR TITLE
Cleans up and refactors the chemistry driver code

### PR DESCRIPTION
**Pull request by singhbalwinder**
A major cleanup of the chemistry driver code. We have removed the state data structure from all the chemistry codes. Pbuf is still there but it is used just for reading a file. The rest of the pbuf instances have been removed.

### CODE REVIEW

**Review by tangshuaiqi — APPROVED** [review]

**Review by jeff-cohere — APPROVED** [review]
👍

**Review by mingxuanwupnnl — APPROVED** [review]

**Comment by mingxuanwupnnl** [comment]
@singhbalwinder I think we need to rebase to resolve the conflicts due to recent merge of gas drydep refactor

File: components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90 [hunk 1]
```diff
@@ -728,123 +507,53 @@ subroutine gas_phase_chemdr(lchnk, ncol, imozart, q, &
...
     enddo
 
-    do m = 1,pcnst
-       n = map2chm(m)
-       if( n > 0 ) then
-          qtend(:ncol,:,m) = qtend(:ncol,:,m) + mmr_tend(:ncol,:,n) 
+    do mm = 1,pcnst
+       nn = map2chm(mm)
+       if( nn > 0 ) then
+          qtend(:ncol,:,mm) = qtend(:ncol,:,mm) + mmr_tend(:ncol,:,nn)
```
Comment (brigaudet) [inline components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90:556]: Very minor -- 'end if' and 'end do' here, and 'end do' line 585.

**Review by brigaudet — APPROVED** [review]
Looks great, just left one very minor comment.

File: components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90 (hunk 1 above)
Comment (singhbalwinder) [reply components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90:556]: I have fixed these and also cleaned up and unified some more codes.

**Comment by singhbalwinder** [comment]
$ ./cs.status.comp_tmp_chem3_mam4_org_v2_baselines -s
comp_tmp_chem3_mam4_org_v2_baselines: 9 tests
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P1x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Pull request merged**